pkg/elastic_service/elastic_handler.go

```diff
@@ -0,0 +1,147 @@
+	put, err := e.Client.Index().
+		Index(consts.Index).
+		BodyString(string(postBody)).
+		Do(*e.Ctx)
+	if err != nil {
+		return "", errors.Wrap(err, err.Error())
+	}
+	return put.Id, err
+}
+func (e *ElasticHandler) Get(id string) (src *models.Book, err error) {
+	get, err := e.Client.Get().
+		Index(consts.Index).
+		Id(id).
+		Do(*e.Ctx)
+	if err != nil {
+		return nil, errors.Wrap(err, err.Error())
+	}
+
+	var getResp models.Book
+	if err = json.Unmarshal(get.Source, &getResp); err != nil {
+		return nil, err
+	}
+	return &getResp, err
+}
+
+func (e *ElasticHandler) Delete(id string) error {
+	_, err := e.Client.Delete().
+		Index(consts.Index).
+		Id(id).
+		Do(*e.Ctx)
+	if err != nil {
+		return errors.Wrap(err, err.Error())
+	}
+	return err
+}
+
+func (e *ElasticHandler) Search(title string, author string, priceStart float64, priceEnd float64) (res []models.Book, err error) {
+	s := ""
+	if title == "" && author == "" && priceEnd == 0 {
+		s = fmt.Sprintf(`{"match_all": {}}`)
+	} else if title == "" && author == "" {
+		s = fmt.Sprintf(`{"range": {"price": {"gte": %f, "lte": %f}}}`, priceStart, priceEnd)
+	} else if title == "" && priceEnd == 0 {
+		s = fmt.Sprintf(`{"constant_score": {"filter": {"bool": {"must":[{"match": {"author": "%s"}}]}}}}`, author)
+	} else if author == "" && priceEnd == 0 {
+		s = fmt.Sprintf(`{"constant_score": {"filter": {"bool": {"must":[{"match": {"title": "%s"}}]}}}}`, title)
+	} else if priceEnd == 0 {
+		s = fmt.Sprintf(`{"constant_score": {"filter": {"bool": {"must":[{"match": {"title": "%s"}},{"match": {"author": "%s"}}]}}}}`, title, author)
+	} else if title == "" {
+		s = fmt.Sprintf(`{"constant_score": {"filter": {"bool": {"must":[{"match": {"author": "%s"}},{"range": {"price": {"gte": %f, "lte": %f} }}]}}}}`, author, priceStart, priceEnd)
+	} else if author == "" {
+		s = fmt.Sprintf(`{"constant_score": {"filter": {"bool": {"must":[{"match": {"title": "%s"}},{"range": {"price": {"gte": %f, "lte": %f} }}]}}}}`, title, priceStart, priceEnd)
+	} else {
+		s = fmt.Sprintf(`{"constant_score": {"filter": {"bool": {"must":[{"match": {"title": "%s"}},{"match": {"author": "%s"}},{"range": {"price": {"gte": %f, "lte": %f} }}]}}}}`, title, author, priceStart, priceEnd)
+	}
+
+	query := elastic.RawStringQuery(s)
+	searchResult, err := e.Client.Search().
+		Index(consts.Index).
+		Query(query).
+		Size(e.maxSizeQuery).
+		Do(context.TODO())
+	if err != nil {
+		return nil, errors.Wrap(err, err.Error())
+	}
+	if searchResult.Hits == nil {
+		return nil, errors.New("expected SearchResult.Hits != nil; got nil")
+	}
+
+	length := len(searchResult.Hits.Hits)
+	res = make([]models.Book, length)
+	for i := 0; i < length; i++ {
+		var src models.Book
+		if err := json.Unmarshal(searchResult.Hits.Hits[i].Source, &src); err != nil {
+			return nil, errors.Wrap(err, err.Error())
+		}
+		res[i] = src
+		res[i].Id = searchResult.Hits.Hits[i].Id
+	}
+	return res, err
+}
+
+func (e *ElasticHandler) Store() (int64, int, error) {
```

Comment:
Can you try on a way of returning more info of the store without breaking the function signature?
```suggestion
func (e *ElasticHandler) StoreInfo() (int64, int, error) {
```